src/hotspot/share/gc/g1/g1BlockOffsetTable.hpp

```diff
@@ -147,9 +131,7 @@ class G1BlockOffsetTablePart {
     }
   }
 
-  void set_for_starts_humongous(HeapWord* obj_top, size_t fill_size);
-
-  void print_on(outputStream* out) PRODUCT_RETURN;
+  void set_for_starts_humongous(HeapRegion* hr, HeapWord* obj_top, size_t fill_size);
```

Comment:
I feel this doesn't belong to BOT. Can probably be dealt with in another ticket.

Reply:
That sounds reasonable. (Should not be done in this PR though.)

(My experience with BOT is that they are almost never corrupted, so doing only checking-after-each-write is enough, sth like `ObjectStartArray::verify_for_block` -- there is possibly little value in verifying BOT in `HeapRegion::verify`. I wonder what others' opinions are.)

Reply:
> I feel this doesn't belong to BOT. Can probably be dealt with in another ticket.

Filed https://bugs.openjdk.org/browse/JDK-8329767 to follow up.

> (My experience with BOT is that they are almost never corrupted, so doing only checking-after-each-write is enough, sth like `ObjectStartArray::verify_for_block` -- there is possibly little value in verifying BOT in `HeapRegion::verify`. I wonder what others' opinions are.)

The verifications of Serial and Parallel BOTs are similar. I agree that G1 should be adjusted. Filed https://bugs.openjdk.org/browse/JDK-8329771 to follow up.